src/puget/color.clj

```diff
@@ -0,0 +1,36 @@
+(ns puget.color
+  "This namespace defines multimethods to add color markup to text.")
+
+;; ## Coloring Multimethods
+
+(defn dispatch
+  "Dispatches to coloring multimethods. Element should be a key from
+  the color-scheme map."
+  [element text options]
+  (and (:print-color options) (:color-markup options)))
+
+
+(defmulti document
+  "Constructs a pretty print document, which may be colored if
+  `:print-color` is true."
+  #'dispatch)
+
+
+(defmulti text
+  "Produces text colored according to the active color scheme. This is mostly
+  useful to clients which want to produce output which matches data printed by
+  Puget, but which is not directly printed by the library. Note that this
+  function still obeys the `:print-color` option."
+  #'dispatch)
+
+
+;; ## No markup when colorless
+
+(defmethod document false
```

Comment:
I'm not sure how I feel about `false` as the "no color" default. Theoretically, people should set `:print-color` to false when they don't want colorizing, not set `:color-markup` to `false`. Since other values for `:color-markup` are keywords, I think `nil` makes a better no-op dispatch, with the bonus that it works if `:color-markup` isn't set in the options for some reason.